contractcourt/chain_watcher.go

```diff
@@ -393,6 +393,96 @@ func isOurCommitment(localChanCfg, remoteChanCfg channeldb.ChannelConfig,
 	return false, nil
 }
 
+// chainSet includes all the information we need to dispatch a channel close
+// event to any subscribers.
+type chainSet struct {
+	// remoteStateNum is the commitment number of the lowest valid
+	// commitment the remote party holds from our PoV.
+	remoteStateNum uint64
+
+	// commitSet includes information pertaining to the set of active HTLCs
+	// on each commitment.
+	commitSet CommitSet
+
+	// remoteCommit points to the current commitment of the remote party.
+	remoteCommit *channeldb.ChannelCommitment
+
+	// localCommit points to our current commitment.
+	localCommit *channeldb.ChannelCommitment
+
+	// remotePendingCommit points to the dangling commitment of the remote
+	// party, if it exists. If there's no dangling commitment, then this
+	// pointer will be nil.
+	remotePendingCommit *channeldb.ChannelCommitment
+}
+
+// newChainSet creates a new chainSet given the current up to date channel
+// state.
+func newChainSet(chanState *channeldb.OpenChannel) (*chainSet, error) {
+	// First, we'll grab the current unrevoked commitments for ourselves
+	// and the remote party.
+	localCommit, remoteCommit, err := chanState.LatestCommitments()
+	if err != nil {
+		return nil, fmt.Errorf("unable to fetch channel state for "+
+			"chan_point=%v", chanState.FundingOutpoint)
+	}
+
+	log.Debugf("ChannelPoint(%v): local_commit_type=%v, local_commit=%v",
+		chanState.ChanType, spew.Sdump(localCommit))
+	log.Debugf("ChannelPoint(%v): remote_commit_type=%v, remote_commit=%v",
+		chanState.ChanType, spew.Sdump(remoteCommit))
+
+	// Fetch the current known commit height for the remote party, and
+	// their pending commitment chain tip if it exist.
+	remoteStateNum := remoteCommit.CommitHeight
+	remoteChainTip, err := chanState.RemoteCommitChainTip()
+	if err != nil && err != channeldb.ErrNoPendingCommit {
+		return nil, fmt.Errorf("unable to obtain chain tip for "+
+			"ChannelPoint(%v): %v",
+			chanState.FundingOutpoint, err)
+	}
+
+	// Now that we have all the possible valid commitments, we'll make the
+	// CommitSet the ChannelArbitrator will need it in order to carry out
+	// its duty.
+	commitSet := CommitSet{
+		HtlcSets: make(map[HtlcSetKey][]channeldb.HTLC),
+	}
+	commitSet.HtlcSets[LocalHtlcSet] = localCommit.Htlcs
+	commitSet.HtlcSets[RemoteHtlcSet] = remoteCommit.Htlcs
+
+	var remotePendingCommit *channeldb.ChannelCommitment
+	if remoteChainTip != nil {
+		remotePendingCommit = &remoteChainTip.Commitment
+		log.Debugf("ChannelPoint(%v): remote_pending_commit_type=%v, "+
+			"remote_pending_commit=%v",
+			chanState.ChanType,
+			spew.Sdump(remoteChainTip.Commitment))
+
+		htlcs := remoteChainTip.Commitment.Htlcs
+		commitSet.HtlcSets[RemotePendingHtlcSet] = htlcs
+	}
+
+	// We'll not retrieve the latest sate of the revocation store so we can
+	// populate the information within the channel state object that we
+	// have.
+	//
+	// TODO(roasbeef): mutation is bad mkay
+	_, err = chanState.RemoteRevocationStore()
```

Comment:
This function is only used in one place (here) and so isn't returning `shachain.Store` in the function pointless if we're ignoring here?

Reply:
also unclear to me what this does... the comment says we retrieve it an populate the information, but we don't

Reply:
Heh this TODO is years old at this point....

The `RemoteRevocationStore` method reads in `RevocationProducer` which is used to tell if this is our commitment or not.